src/collections.ts

```diff
@@ -0,0 +1,147 @@
+import * as collections from "@opencreek/deno-std-collections"
+import { apply, extendProtoype } from "./extendPrototype"
+
+type Collections = typeof collections
+
+function filterNotNullish<T>(this: Array<T>): Array<NonNullable<T>> {
+  return this.filter((it): it is NonNullable<T> => it != undefined)
+}
+
+function extendArray<
+  KEY extends {
+    [P in keyof Collections]: Collections[P] extends (
+      first: any[],
+      ...args: any
+    ) => any
+      ? P
+      : never
+  }[keyof Collections]
+>(key: KEY): void {
+  extendProtoype(Array, apply(collections[key]), key)
+}
+
+extendProtoype(Array, filterNotNullish, "filterNotNullish")
+
+extendArray("associateBy")
+extendArray("associateWith")
+extendArray("chunk")
+extendArray("distinct")
+extendArray("distinctBy")
+extendArray("dropLastWhile")
+extendArray("dropWhile")
+extendArray("findLast")
+extendArray("findLastIndex")
+extendArray("findSingle")
+extendArray("firstNotNullishOf")
+extendArray("groupBy")
+extendArray("intersect")
+extendArray("mapNotNullish")
+extendArray("maxBy")
+extendArray("maxOf")
+extendArray("maxWith")
+extendArray("minBy")
+extendArray("minOf")
+extendArray("minWith")
+extendArray("partition")
+extendArray("permutations")
+extendArray("runningReduce")
+extendArray("sample")
+extendArray("slidingWindows")
+extendArray("sortBy")
+extendArray("sumOf")
+extendArray("takeLastWhile")
+extendArray("takeWhile")
+extendArray("takeWhile")
+extendArray("union")
+extendArray("unzip")
+extendArray("withoutAll")
+extendArray("zip")
```

Comment:
Nitpick: DRYing this up with an array and a `.forEach` might be nice. Vim Macro ftw?^^